engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/FetchCachedResultsActor.scala

```diff
@@ -0,0 +1,41 @@
+package cromwell.engine.workflow.lifecycle.execution.callcaching
+
+import akka.actor.{Actor, ActorLogging, Props}
+import cromwell.Simpletons._
+import cromwell.core.{DoesNotExpectMessages, JobOutputs}
+import cromwell.core.simpleton.WdlValueBuilder
+import cromwell.database.CromwellDatabase
+import cromwell.database.sql.MetaInfoId
+import cromwell.engine.workflow.lifecycle.execution.callcaching.FetchCachedResultsActor.{CachedOutputLookupFailed, CachedOutputLookupSucceeded}
+import wdl4s.TaskOutput
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
+
+object FetchCachedResultsActor {
+  def props(metaInfoId: MetaInfoId, taskOutputs: Seq[TaskOutput]): Props =
+    Props(new FetchCachedResultsActor(metaInfoId, taskOutputs))
+
+  sealed trait CachedResultResponse
+  case class CachedOutputLookupFailed(metaInfoId: MetaInfoId, failure: Throwable) extends CachedResultResponse
+  case class CachedOutputLookupSucceeded(jobOutputs: JobOutputs) extends CachedResultResponse
+}
+
+
+class FetchCachedResultsActor(metaInfoId: MetaInfoId, taskOutputs: Seq[TaskOutput])
+  extends Actor with ActorLogging with DoesNotExpectMessages with CromwellDatabase {
+  {
+    implicit val ec: ExecutionContext = context.dispatcher
+
+    val callCache = new CallCache(CromwellDatabase.databaseInterface)
+    val replyTo = context.parent
+
+    callCache.fetchCachedResult(metaInfoId) onComplete {
+      case Success(result) =>
+        val simpletons = result.resultSimpletons map toSimpleton
+        val jobOutputs = WdlValueBuilder.toJobOutputs(taskOutputs, simpletons)
+        replyTo ! CachedOutputLookupSucceeded(jobOutputs)
```

Comment:
I'm pretty sure `pipeTo` was only added to akka in the first place to avoid sender-in-closure bugs. 
OTOH, I personally like the style and intend to use it more where appropriate